lib/net/nforceif/src/sys_arch.c

```diff
@@ -347,31 +350,26 @@ sys_sem_new(struct sys_sem **sem, u8_t count)
 u32_t
 sys_arch_sem_wait(struct sys_sem **s, u32_t timeout)
 {
-    u32_t start, time_needed = 0;
     struct sys_sem *sem;
     LWIP_ASSERT("invalid sem", (s != NULL) && (*s != NULL));
     sem = *s;
-    int done;
-
-    start = sys_now();
-
-    for (done=0; !done;) {
-        NtWaitForSingleObject(sem->handle, FALSE, NULL);
 
-        if (sem->c > 0) {
-            sem->c -= 1;
-            done = 1;
-        }
-
-        NtReleaseSemaphore(sem->handle, 1, NULL);
+    LARGE_INTEGER timeout_large;
+    PLARGE_INTEGER timeout_ptr;
+    if (timeout) {
+        timeout_large.QuadPart = (long long int)timeout * -10000;
+        timeout_ptr = &timeout_large;
+    } else {
+        timeout_ptr = NULL;
+    }
 
-        if (timeout > 0) {
-            time_needed = sys_now()-start;
-            if (time_needed >= timeout) return SYS_ARCH_TIMEOUT;
-        }
+    NTSTATUS status = NtWaitForSingleObject(sem->handle, FALSE, timeout_ptr);
+    if (status == STATUS_TIMEOUT) {
+        return SYS_ARCH_TIMEOUT;
     }
+    assert(status == STATUS_SUCCESS);
 
-    return (u32_t)time_needed;
+    return 0;
```

Comment:
*(This is mostly a note to myself, that can be answered by anyone)*

I still have to check the lwip docs, but why is this constant 0 now, when it used to return how much time was spent? How is this return value used?

Also what is `SYS_ARCH_TIMEOUT`? A duration? in what unit? or a magic constant?

Reply:
Oops, screwed that up during rewrite. It's supposed to return the milliseconds spent waiting, which it does now.

`SYS_ARCH_TIMEOUT` is just some constant (`0xffffffff`) to signal timeouts.